assets/src/blocks/HubspotForm/HubspotFormBlock.js

```diff
@@ -0,0 +1,159 @@
+import ReactDOMServer from 'react-dom/server';
+import { RawHTML, Fragment, renderToString } from '@wordpress/element';
+import { Tooltip } from '@wordpress/components';
+import { HubspotFormEditor } from './HubspotFormEditor';
+import { HubspotFormFrontend } from './HubspotFormFrontend';
+const { __ } = wp.i18n;
+const { registerBlockType } = wp.blocks;
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+
+const BLOCK_NAME = 'planet4-blocks/hubspot-form';
+
+const getStyleLabel = (label, help) => (
+  (help)
+    ? <Tooltip text={help}>
+        <span>{label}</span>
+      </Tooltip>
+    : label
+);
+
+export const registerHubspotFormBlock = () => {
+  return registerBlockType(BLOCK_NAME, {
+    title: 'Hubspot Form (beta)',
+    icon: 'feedback',
+    category: 'planet4-blocks-beta',
+    supports: {
+      multiple: false,
+      html: false,
+    },
+    attributes: {
+      block_title: {
+        type: 'string',
+      },
+      block_text: {
+        type: 'string',
+      },
+      block_background_image_id: {
+        type: 'integer',
+      },
+      block_background_image_url: {
+        type: 'string',
+      },
+      cta_text: {
+        type: 'string',
+      },
+      cta_link: {
+        type: 'string',
+      },
+      cta_new_tab: {
+        type: 'boolean',
+        default: false,
+      },
+      form_title: {
+        type: 'string',
+      },
+      form_description: {
+        type: 'string',
+      },
+      hubspot_shortcode: {
+        type: 'string',
+      },
+      hubspot_thankyou_message: {
+        type: 'string',
+      },
+      enable_custom_hubspot_thankyou_message: {
+        type: 'boolean',
+        default: true,
+      },
+      version: {
+        type: 'integer',
+        default: 1,
+      },
+    },
+    styles: [
+      {
+        name: 'imageFullWidth',
+        label: getStyleLabel(
+          __('Image full width', 'planet4-blocks-backend'),
+          __('Image full width version', 'planet4-blocks-backend'),
+        ),
+        isDefault: true,
+      },
+    ],
+
+    /**
+      The belowed function works correctly without using innerBlocks.
+    */
+    // edit: (props) => {
+    //   return (
+    //     <Fragment>
+    //       <HubspotFormEditor {...props} />
+    //     </Fragment>
+    //   )
+    // },
+
+    /**
+      The belowed function works correctly using innerBlocks.
```

Comment:
As discussed on Slack, InnerBlocks doesn't allow us the behavior we want, because it only works well if the block that contains it is also saved. In our case the block needs to be rendered, or hydrated. Neither is compatible yet with InnerBlocks.

However we can achieve the same goal (using a block editor for the thank you message) in a much simpler way: By creating a separate block for the thank you message, which is hidden by default on the page, and then shown by the form submit handler